mlu_op.h

```diff
@@ -14523,6 +14523,132 @@ mluOpLgamma(mluOpHandle_t handle,
             const mluOpTensorDescriptor_t y_desc,
             void *y);
 
+/*!
+ * @brief Calculates the size of the workspace required for the LU decomposition and initializes a workspace pointer.
+ * This function must be called before performing LU decomposition using mluOpSgetrf2.
+ *
+ * @param[in] handle
+ * Handle to a Cambricon MLUOP context that is used to manage MLU devices and
+ * queues in the deformable convolution backward data operation. For detailed information,
+ * see ::mluOpHandle_t.
+ *
+ * @param[in] input_desc
+ * The descriptor for the input tensor for which the LU decomposition will be performed.
+ *
+ * @param[out] workspace_size
+ * Pointer to a variable where the size of the required workspace will be stored.
+ *
+ * @par Return
+ * - ::MLUOP_STATUS_SUCCESS if the workspace size is successfully calculated and the workspace is successfully
+ * allocated,
+ * - ::MLUOP_STATUS_EXECUTION_FAILED if there are issues during the calculation or memory allocation.
+ *
+ * @par Data Type
+ * - None.
+ *
+ * @par Data Layout
+ * - The supported combinations of data types are shown below:
+ * - size_t(size)
+ *
+ * @par Scale Limitation
+ * - The dimension of input tensor must be either 2, 3 or 4.
+ *
+ * @par API Dependency
+ * - The allocated extra workspace should be passed to ::mluOpSgetrf2 to perform the LU operation.
+ *
+ * @par Note
+ * - None.
+ *
+ * @par Example
+ * - None.
+ *
+ * @par Reference
+ * - None.
+ */
+mluOpStatus_t MLUOP_WIN_API
+mluOpGetSgetrf2WorkspaceSize(mluOpHandle_t handle, const mluOpTensorDescriptor_t input_desc, size_t *workspace_size);
+/*!
+ * @brief Computes the LU decomposition of a matrix using the input tensor descriptor \p input_desc and writes the
+ result to the output tensor descriptor \p output_desc.
+
+ * @param[in] handle
+ *   Handle to a Cambricon MLUOP context that is used to manage MLU devices and
+ *   queues in the deformable convolution backward data operation. For detailed information,
+ *   see ::mluOpHandle_t.
+ *
+ * @param[in] x_desc
+ *   The descriptor of the input tensor. For detailed information,
+ *   see ::mluOpTensorDescriptor_t.
+ *
+ * @param[in] x
+ *   Pointer to the MLU memory that stores the input tensor.
+ *
+ * @param[in] y_desc
+ *   The descriptor of the output tensor. For detailed information,
+ *   see ::mluOpTensorDescriptor_t.
+ *
+ * @param[out] y
+ *   Pointer to the MLU memory that stores the output tensor.
+ *
+ * @param[in, out] workspace
+ * Pointer to the MLU memory that is used as an extra workspace for the
+ * ::mluOpSgetrf2.
+ *
+ * @param[in, out] ipiv
+ * INTEGER array, dimension (m);
+ * The pivot indices; row i of the matrix was interchanged with row IPIV(i)
```

Comment:
```suggestion
 * An integer array, dimension (m);
 * The pivot indices; row i of the matrix was interchanged with row IPIV(i).
```

需要优化，说明ipiv的含义。写成完整的句子：这里dimension跟pivot indices是啥关系。

Reply:
已修改